src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateEqualsMethod.cs

```diff
@@ -322,7 +322,7 @@ private static bool ImplementsIEquatable(ITypeSymbol memberType, INamedTypeSymbo
             if (iequatableType != null)
             {
                 // It's correct to throw out nullability here -- if you have a field of type Foo? and it implements IEquatable, it's still implementing IEquatable<Foo>.
-                var constructed = iequatableType.Construct(memberType.WithoutNullability());
+                var constructed = iequatableType.Construct(memberType.WithNullableAnnotation(NullableAnnotation.None));
                 return memberType.AllInterfaces.Contains(constructed);
```

Comment:
❔ Won't this `Contains` use `Equals`, which is a comparison that ignores nullability?

The important consideration here is the fact that a value type `V` will just implement `IEquatable<V>`, but a reference type `R` can implement _either_ `IEquatable<R>` or `IEquatable<R?>`. The latter is typically preferred because it better supports multitargeting scenarios.

Reply:
Ah, right: prior to the compiler wrapping, this would have had to strip the wrapper or the Contains wouldn't work because the comparison would have outright failed since our symbols were "unknown" according to it's comparer. Removing the WithNullable() annotation and being explicit about the comparer ignoring nullability should now work.

Reply:
Comment updated.